summingbird-online/src/main/scala/com/twitter/summingbird/online/executor/AsyncBase.scala

```diff
@@ -107,13 +136,15 @@ abstract class AsyncBase[I, O, S, D, RC](maxWaitingFutures: MaxWaitingFutures, m
     if (pendingFuturesCount > maxWaitingFuturesCount) {
       // Too many futures waiting, let's clear.
       outstandingFutures.dequeueAll(_.isDefined)
-      val toForce = outstandingFutures.trimTo(maxWaitingFuturesCount).toIndexedSeq
```

Comment:
Could you explain a bit more on "one future could not wait too long".  The wait would have timed out with MaxFutureWaitTime, are you referring to that. In the new algo it can stay in the queue forever but that should be rare, futures would almost always have timeouts, if they don't then that's likely a bug.